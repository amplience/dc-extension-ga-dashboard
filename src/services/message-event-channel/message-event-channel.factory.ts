import { ClientConnection } from 'message-event-channel';

interface connectionOptions {
  targetOrigin?: string;
  onload?: boolean;
  timeout?: number | boolean;
  debug?: boolean;
  connectionTimeout?: number | boolean;
  clientInitiates?: boolean;
}

const createConnection = (options?: connectionOptions): ClientConnection => {
  return new ClientConnection(options);
};

export default createConnection;
