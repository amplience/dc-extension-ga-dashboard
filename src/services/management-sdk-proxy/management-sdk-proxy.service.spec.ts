import ManagementSDKProxyService, {
  ManagementSDKProxy,
} from './management-sdk-proxy.service';
import { ClientConnection } from 'message-event-channel';

jest.mock('message-event-channel');

describe('management sdk proxy service - Using connection', () => {
  let connection: ClientConnection;
  beforeEach(() => {
    connection = new ClientConnection();
    (connection.request as jest.Mock).mockReturnValue({
      apiUrl: 'https://mock.api.amplience.net/v2/content/',
      hubId: 'hubId',
    });
  });
  it('should return a new client', async () => {
    const sdkProxyService = new ManagementSDKProxyService(connection);
    expect(await sdkProxyService.getClient()).toMatchSnapshot();
    expect(sdkProxyService.hubId).toEqual('hubId');
  });
});

describe('management sdk proxy service - Standalone mode', () => {
  let connection: ClientConnection;
  let sdkProxyService: ManagementSDKProxy;
  beforeEach(() => {
    connection = new ClientConnection();
    sdkProxyService = new ManagementSDKProxyService(connection);
    (sdkProxyService as ManagementSDKProxy & {
      isStandalone: () => string;
    }).isStandalone = () => '1';
  });
  it('should return a new client', async () => {
    expect(await sdkProxyService.getClient()).toMatchSnapshot();
    expect(sdkProxyService.hubId).toEqual('__HUB_ID__');
  });
});
