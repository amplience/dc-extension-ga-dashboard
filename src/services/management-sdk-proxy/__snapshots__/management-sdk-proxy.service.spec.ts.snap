// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`management sdk proxy service - Standalone mode should return a new client 1`] = `
DynamicContent {
  "client": DefaultHalClient {
    "baseUrl": "__API_URL__",
    "httpClient": AxiosHttpClient {
      "client": [Function],
      "config": Object {},
    },
    "tokenProvider": OAuth2Client {
      "authUrl": "__AUTH_URL__",
      "clientCredentials": Object {
        "client_id": "__CLIENT_ID__",
        "client_secret": "__CLIENT_SECRET__",
      },
      "httpClient": AxiosHttpClient {
        "client": [Function],
        "config": Object {},
      },
    },
  },
  "contentItems": Object {
    "get": [Function],
  },
  "contentRepositories": Object {
    "get": [Function],
  },
  "contentTypeSchemas": Object {
    "get": [Function],
    "getByVersion": [Function],
  },
  "contentTypes": Object {
    "get": [Function],
  },
  "editions": Object {
    "get": [Function],
  },
  "events": Object {
    "get": [Function],
  },
  "folders": Object {
    "get": [Function],
  },
  "hubs": Object {
    "get": [Function],
    "list": [Function],
  },
  "snapshots": Object {
    "get": [Function],
  },
}
`;

exports[`management sdk proxy service - Using connection should return a new client 1`] = `
DynamicContentProxy {
  "client": DefaultHalClient {
    "baseUrl": "https://mock.api.amplience.net/v2/content/",
    "httpClient": DynamicContentClientProxy {
      "connection": ClientConnection {
        "addBeforeUnloadEvent": [MockFunction],
        "init": [MockFunction],
        "isClient": [MockFunction],
        "listenForHandshake": [MockFunction],
        "messageHandler": [MockFunction],
      },
    },
    "tokenProvider": Object {
      "getToken": [Function],
    },
  },
  "contentItems": Object {
    "get": [Function],
  },
  "contentRepositories": Object {
    "get": [Function],
  },
  "contentTypeSchemas": Object {
    "get": [Function],
    "getByVersion": [Function],
  },
  "contentTypes": Object {
    "get": [Function],
  },
  "editions": Object {
    "get": [Function],
  },
  "events": Object {
    "get": [Function],
  },
  "folders": Object {
    "get": [Function],
  },
  "hubs": Object {
    "get": [Function],
    "list": [Function],
  },
  "snapshots": Object {
    "get": [Function],
  },
}
`;
